driver/compenv.ml

```diff
@@ -587,14 +631,75 @@ type deferred_action =
 let c_object_of_filename name =
   Filename.chop_suffix (Filename.basename name) ".c" ^ Config.ext_obj
 
+let check_ir name =
+  let check_suffix () =
+    let ext = Filename.extension name in
+    let ext_len = String.length ext in
+    if ext_len <= 0 then
+      None
+    else begin
+      List.find_opt (fun ir ->
+          let s = Compiler_ir.extension ir in
+          (* Check whether [ext] starts with [s]. It is not an exact match,
+             to distinguish different passes on the same IR. *)
+          let s_len = String.length s in
+          s_len <= ext_len && s = String.sub ext 0 s_len)
```

Comment:
I suggest to put this code in a function in `Compiler_ir`, just to make the logic more obvious here. For instance:

```ocaml
(** for instance: [extract_extension_with_subpass "foo.cmir-linear123" = Some (Linear, "123") *)
val extract_extension_with_subpass : string -> (t * string) option
```

Reply:
done